### ISSUE TITLE
TODO for neural network approximation of ODE

### ISSUE THREAD

**Issue by mikeaalv** [artedison/NeuralSimODE#2]
The network method has worked for comparably small ODE set (dimension=4), and there is obvious trouble in making it work for larger ODE set. Here I list a few things that has been tried or might work (but haven't been tried).

- [x] rennet

- [x] optimizer scheduler 

- [ ] automatic hyper-parameter searching (such as optuna)

- [ ] fix memory problem to increase batch size and speed

- [x] gru

- [x] gru-mlp (a modified version of gru)

- [x] diff-add-cell (a designed RNN structure)

- [ ] ideas for improve fitting ??
